### ISSUE TITLE
Must not complain about 'new style hashes' if symbol is a string.

### ISSUE THREAD

**Issue by sleekweasel** [bbatsov/rubocop#2474]
C: Use the new Ruby 1.9 hash syntax.
    :'Allow profile sharing' => 'profile_can_be_shared_by_others'
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^

is an incorrect complaint, since

ruby -e "{'symbol key with space': 'value'}"
-e:1: syntax error, unexpected ':', expecting =>
{'symbol key with space': 'value'}
                         ^
-e:1: syntax error, unexpected '}', expecting end-of-input

**Comment by alexdowad** [comment]
`ruby -v`?

**Comment by sleekweasel** [comment]
2.1.5 or 2.2.3 - I'll have to check when I get back into the office.

**Comment by latortuga** [comment]
Worth noting that in your example, your hash key is a symbol, not a string. It's probably failing because you specified a symbol key but didn't use the hash rocket syntax. Obviously the hash rocket syntax is required in this case so perhaps the rule should check the key to ensure is both a symbol and that it is allowed to use the ruby19 syntax (in this case you cannot).

**Comment by alexdowad** [comment]
```ruby
22:10 ~prog/Ruby/rubocop % rbenv shell 2.2.2                                   [master]
22:10 ~prog/Ruby/rubocop % pry                                                 [master]
[1] pry(main)> { "hi there": 1 }
=> {:"hi there"=>1}
[2] pry(main)> {'symbol key with space': 'value'}
=> {:"symbol key with space"=>"value"}
[3] pry(main)> RUBY_VERSION
=> "2.2.2"
```

This bug report is incorrect and should be closed.

**Comment by jonas054** [comment]
I agree. @sleekweasel I think you ran `rubocop` under Ruby 2.2 or later, but the `ruby -e` command on an earlier version.
